src/backend/executor/spi.c

```diff
@@ -2093,6 +2093,7 @@ _SPI_execute_plan(SPIPlanPtr plan, ParamListInfo paramLI,
 	ErrorContextCallback spierrcontext;
 	CachedPlan *cplan = NULL;
 	ListCell   *lc1;
+	bool orig_gp_enable_gpperfmon = gp_enable_gpperfmon;
```

Comment:
I think you should store gp_enable_gpperfmon not in this place, but below, before PG_TRY, as done in the fmgr_sql function.

Reply:
Moved